lib/tesla/middleware/telemetry.ex

```diff
@@ -0,0 +1,37 @@
+defmodule Tesla.Middleware.Telemetry do
+  @behaviour Tesla.Middleware
+
+  @moduledoc """
+  Send the request duration and meta-information through telemetry.
+
+  ### Example usage
+  ```
+  defmodule TelemetryHandler do
+    use GenServer
+
+    def start_link(_, _) do
+      :telemetry_app.start(nil, nil)
+      :telemetry.attach(
+        "tesla-telemetry",
+        [:tesla, :telemetry, :traffic],
+        fn ([:tesla, :telemetry, :traffic], time, meta, _config) -> # Do sth end,
+        nil)
+    end
+  end
+
+  defmodule MyClient do
+    use Tesla
+
+    plug Tesla.Middleware.Telemetry
+
+  end
+  ```
+  """
+
+  @doc false
+  def call(env, next, _opts) do
+    {time, res} = :timer.tc(Tesla, :run, [env, next])
+    :telemetry.execute([:tesla, :telemetry, :traffic], time, Enum.into([res], %{}))
```

Comment:
I'm not sure about the event name, maybe we should use `[:tesla, :request, :done]` or even `[:tesla, :request]`? The `:telemetry` part seems redundant.

Reply:
👍 for `[:tesla, :request]`. `telemetry`, `traffic`, and `done` seems redundant.